### PR TITLE
Remove dead code and deprecated syntax

### PR DESCRIPTION
**Pull request by yebblies**

### CODE REVIEW

**Comment by yebblies** [comment]
Thanks, will do.

**Comment by yebblies** [comment]
Auto-merge toggled on

**Comment by schuetzm** [comment]
This PR introduced a regression:
https://issues.dlang.org/show_bug.cgi?id=15253

**Comment by schuetzm** [comment]
Just from looking at the diff, I suspect it's caused by the change in dtemplate.d.

File: src/dtemplate.d [hunk 1]
```diff
@@ -1688,7 +1688,7 @@ public:
                 default:
                     goto Lnomatch;
                 }
-                ++argi;
+                assert(0);
```
Comment (WalterBright) [inline src/dtemplate.d]: this change looks suspicious

Comment (yebblies) [inline src/dtemplate.d]: If it's wrong then there's a compiler bug in dead code detection.

Comment (WalterBright) [inline src/dtemplate.d]: The easiest way to find out what caused the regression is to back all these changes out, then add them one by one until the regression happens.

**Comment by MartinNowak** [comment]
This PR looks good (and reverting it doesn't fix the issue).

**Pull request merged**